pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -142,7 +146,9 @@ let
     (enableFeature enableExecutableProfiling (if versionOlder ghc.version "8" then "executable-profiling" else "profiling"))
     (enableFeature enableSharedLibraries "shared")
     (optionalString (versionAtLeast ghc.version "7.10") (enableFeature doCoverage "coverage"))
-    (optionalString (versionOlder "8.4" ghc.version) (enableFeature enableStaticLibraries "static"))
+    # --enable-static does not work on windows. This is a bug in GHC.
+    # --enable-static will pass -staticlib to ghc, which only works for mach-o and elf.
+    (optionalString (!hostPlatform.isWindows && versionOlder "8.4" ghc.version) (enableFeature enableStaticLibraries "static"))
```

Comment:
If someone requests static builds for a given platform, then it feels like poor style to just ignore the setting and to silently build shared libraries instead. I'd prefer to see this particular limitation expressed as an assertion instead.

Reply:
I'll amend the PR, to 
```
enableStaticLibraries ? !hostPlatform.isWindows
```
and a
```
assert (enableStaticLibraries && hostPlatform.isWindows) "can't use `--enable-static` on windows."
```

The reason why this does not work is not that we can't built static objects, we can, but we can't use
`-staticlib` on ghc on windows. `-staticlib` rolls all dependencies into a combined archive. While this
would work on windows if we used gnu ar and MRI script, GHC can't rely on gnu ar, and as such has
a quck archive concatenation module for gnu and bsd archives only.

Reply:
On Nix we do indeed provide GNU ar always, so it would be nice to optionally configure GHC to expect GNU binutils just as one can configure GCC for the same reasons.